Public/Src/Cache/ContentStore/Vfs/VfsProgram.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BuildXL.ToolSupport;
+
+namespace BuildXL.Cache.ContentStore.Vfs
+{
+    using System.Diagnostics.ContractsLight;
+    using BuildXL.Cache.ContentStore.Interfaces.FileSystem;
+    using static CommandLineUtilities;
+
+    public class VfsProgram : ToolProgram<VfsCasConfiguration>
+    {
+        public VfsProgram()
+            : base("Bvfs")
+        {
+        }
+
+        public static int Main(string[] arguments)
+        {
+            try
+            {
+                return new VfsProgram().MainHandler(arguments);
+            }
+            catch (InvalidArgumentException e)
+            {
+                Console.Error.WriteLine("Execution error: " + (e.InnerException ?? e).Message);
+            }
+
+            return -1;
+        }
+
+        public override int Run(VfsCasConfiguration arguments)
+        {
+            var runner = new VfsCasRunner();
+            runner.RunAsync(arguments).GetAwaiter().GetResult();
+            return 0;
+        }
+
+        public override bool TryParse(string[] rawArgs, out VfsCasConfiguration arguments)

```

Comment:
>TryParse [](start = 29, length = 8)

Why not use CLAP?

Reply:
I'll probably end up using that just because I don't want to write a help text infrastructure. The reason I avoided it was because CLAP doesn't support key value pair and full path parsing.

---
In reply to: [280869902](https://github.com/Microsoft/BuildXL/pull/249#discussion_r280869902) [](ancestors = 280869902)